addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/impl/OsioWitClientImpl.java

```diff
@@ -0,0 +1,159 @@
+package io.fabric8.launcher.osio.client.impl;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.URI;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import javax.ws.rs.core.HttpHeaders;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.launcher.base.http.HttpException;
+import io.fabric8.launcher.osio.client.api.BadTenantException;
+import io.fabric8.launcher.osio.client.api.ImmutableNamespace;
+import io.fabric8.launcher.osio.client.api.ImmutableSpace;
+import io.fabric8.launcher.osio.client.api.ImmutableTenant;
+import io.fabric8.launcher.osio.client.api.ImmutableUserInfo;
+import io.fabric8.launcher.osio.client.api.Namespace;
+import io.fabric8.launcher.osio.client.api.OsioWitClient;
+import io.fabric8.launcher.osio.client.api.Space;
+import io.fabric8.launcher.osio.client.api.Tenant;
+import io.fabric8.launcher.osio.client.api.UserInfo;
+import io.fabric8.launcher.osio.steps.WitSteps;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import static io.fabric8.launcher.base.http.ExternalRequest.execute;
+import static io.fabric8.launcher.base.http.ExternalRequest.executeAndParseJson;
+import static io.fabric8.launcher.base.identity.IdentityFactory.createFromToken;
+import static io.fabric8.launcher.base.identity.IdentityHelper.removeBearerPrefix;
+import static io.fabric8.launcher.osio.OsioConfigs.getWitUrl;
+import static io.fabric8.utils.URLUtils.pathJoin;
+import static okhttp3.MediaType.parse;
+import static okhttp3.RequestBody.create;
+
+public final class OsioWitClientImpl implements OsioWitClient {
+    private static final Logger LOG = Logger.getLogger(WitSteps.class.getName());
+
+    private final String authorizationHeader;
+
+    public OsioWitClientImpl(final String authorizationHeader) {
+        this.authorizationHeader = authorizationHeader;
+    }
+
+    @Override
+    public Tenant getTenant() {
+        return ImmutableTenant.builder()
+                .userInfo(getUserInfo())
+                .namespaces(getNamespaces())
+                .identity(createFromToken(removeBearerPrefix(authorizationHeader)))
+                .build();
+    }
+
+    @Override
+    public UserInfo getUserInfo() {
+        final Request userInfoRequest = newAuthorizedRequestBuilder("/api/user").build();
+        return executeAndParseJson(userInfoRequest, OsioWitClientImpl::readUserInfo)
+                .orElseThrow(() -> new BadTenantException("UserInfo not found"));
+    }
+
+    @Override
+    public List<Namespace> getNamespaces() {
```

Comment:
This method can be private. I don't think we'll be using it outside getTenant

Reply:
Yes